### PR TITLE
SOLR-13734 JWTAuthPlugin to support multiple issuers

### PR DESCRIPTION
**Pull request by janhoy**
# Description

See https://issues.apache.org/jira/browse/SOLR-13734

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I am authorized to contribute this code to the ASF and have removed any code I do not have a license to distribute.
- [x] I have developed this patch against the `master` branch.
- [x] I have run `ant precommit` and the appropriate test suite.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Comment by janhoy** [comment]
Reviewers: This PR includes changes from #852 and therefore looks bigger than it is. A diff between the two can be seen here:
https://github.com/cominvent/lucene-solr/compare/SOLR-13713-multiple-jwks...cominvent:SOLR-13734-jwt-multiple-issuers

**Comment by gerlowskija** [comment]
Cross-posting this comment from JIRA.  (When JIRAs have github PR's what's the appropriate place for higher-level, non-line-specific review comments?  Is there a consensus on this?  Is one more discoverable than the other?)

----

What's the purpose of the distinction between the "primary" issuer and the secondary issuers under the issuers key? I imagine the "primary" issuer is just kept around for back-compat purposes? If it's just for back-compat, I understand it needs to be done, but it's a shame. The JSON would be easier to understand (IMO) if all of the issuer-specific properties lived only under the new `issuers` key post-9.0. Would it be possible to deprecate iss, wellKnownUrl etc outside of the issuers hash in this PR? That'd make it easier to get to the cleaner alternative at some point down the road...

**Comment by janhoy** [comment]
> When JIRAs have github PR's what's the appropriate place for higher-level, non-line-specific review comments? Is there a consensus on this? Is one more discoverable than the other?

Good question. I'd hope we could move totally to GitHub issues+pr and scrap JIRA. But for now my thought is that if it is a big PR that will attract much comments in GitHub perhaps it is best to keep general comments in the PR as well, to keep all in one place, and then update the JIRA once in a while with general progress, i.e. (planning to merge in 3 days) to attract more attention.

Thanks for your review. See JIRA for comments :) 

> Would it be possible to deprecate iss, wellKnownUrl etc outside of the issuers hash in this PR?

I'll wait until both the "REST API issuers support" and "Admin UI choose issuer to log in with" are solved, so deprecation will not happen in this PR.

File: solr/solr-ref-guide/src/major-changes-in-solr-9.adoc [hunk 1]
```diff
@@ -44,3 +44,5 @@ A thorough review of the list in Major Changes in Earlier 8.x Versions as well a
 === Authentication & Security Changes in Solr 9
 
 * BasicAuthPlugin property 'blockUnknown' now defaults to 'true'. This change is backward incompatible. If you need the pre-9.0 default behavior, you need to explicitly set 'blockUnknown':'false' in security.json.
+
+* JWTAuthPlugin configuration option `requireSub` is no longer needed and will cause an error if used.
```
Comment (janhoy) [inline solr/solr-ref-guide/src/major-changes-in-solr-9.adoc]: Note to self: This is not part of this PR, must remember to remove requireSub totally in master after merge to branch_8x
